test/functional/view-models.js

```diff
@@ -60,6 +60,41 @@ describe('View all models', function () {
         });
     });
 
+    it('adds a new item to the model', () => {
+        mockServer.expect({
+            method: 'post',
+            endpoint: 'item',
+            json: {
+                modelAddress: models[0].address
+            },
+            response: {
+                status: 'success',
+                data: {
+                    address: 'iGwEZVvgu',
+                    modelName: 'Resistor'
+                }
+            }
+        });
+        mockServer.expect({
+            method: 'get',
+            endpoint: 'model/all',
+            response: {
+                status: 'success',
+                data: {
+                    models
+                }
+            }
+        });
+        return app.client.click('.btnAddItemToModel:nth-of-type(1)').then(() => {
+            return app.client.waitForVisible('.modal', 5000);
+        }).then(() => {
+            return app.client.click('.modal .modal-content button[type="button"]');
+        }).then(()=> {
+            return app.client.waitForVisible('.toast', 1000);
+        }).then(() => {
+            mockServer.validate();
```

Comment:
But that page is never left, so I don't see why React would re-render multiple components when only a single component's state has changed. I expect that only the `<Model>` component that you clicked the "Add" button on would be re-rendered—and in the same place in the DOM. Therefore, I expect `.model:nth-of-type(1)` (analogous to the selector used to clicking "Add") to be the same model that just had an item add, and have an increment in Quantity.

You're right it's not the client's responsibility to do `quantity++`, but it _is_ the client's responsibility to update it's display to reflect the actions the user took. We already have a UI bug where an item is still displayed after it has been deleted. This assert would prevent something like that regressing.